awesome_owl/static/src/counter/counter.js

```diff
@@ -0,0 +1,19 @@
+/** @odoo-module **/
+
+import { useState, Component } from "@odoo/owl";
+
+export class Counter extends Component {
+    static template = "awesome_owl.counter";
+    static props = { onChange: { type: Function, optional: true } }
+
+    setup() {
+        this.state = useState({ value: 0 });
+    }
+
+    increment() {
+        this.state.value++;
+        if (this.props.onChange) {
+            this.props.onChange()
+        }
```

Comment:
tip: equivalent syntactic sugar:
```suggestion
        this.props.onChange?.()
```
(arguably harder to read so personal choice)